lib/eslint/eslint.js

```diff
@@ -313,7 +327,18 @@ async function loadFlatConfigFile(filePath) {
         delete require.cache[filePath];
     }
 
-    const config = (await import(fileURL)).default;
+    let config;
+
+    if (isFileTS(filePath)) {
```

Comment:
Because Node.js is the only runtime that can't load TypeScript natively, maybe we can also check that `globalThis.Deno` and `globalThis.Bun` are undefined before loading `importx`?

Reply:
With register/module-loader like `ts-node` or `tsx` (the cli), or testing frameworks like Vitest, Node.js can load TypeScript - the goal for `importx` is to cover those details, where ESLint or other users can be agnostic to runtime like Deno and Bun, while also could automatically work if we have yet another runtime in the future.

`importx` already doing lazying loading internally - so importing `importx` itself should be fairly lightweight.

Reply:
Just to make sure I'm understanding, you're saying `importx` will already check for Deno and Bun and not do anything additional in that case?

Reply:
Yes, exactly.

Reply:
I tried to install `importx` as a dev dependency on Deno and it still downloaded a bunch of packages, including `jiti`, `tsx` and `typescript`. It seems unnecessary to require users to install all those packages if their runtime handles TS files out of teh box.

Reply:
Oh yeah that's a good point. In that case, I think a line to check the `Deno` and `Bun` in ESLint won't hurt, yes - Will do.